src/PromiseInterface.php

```diff
@@ -91,8 +93,9 @@ public function catch(callable $onRejected): PromiseInterface;
      *     ->finally('cleanup');
      * ```
      *
-     * @param callable $onFulfilledOrRejected
-     * @return PromiseInterface
+     * @template TReturn of PromiseInterface<T>|T
+     * @param callable(T=): TReturn $onFulfilledOrRejected
```

Comment:
The `$onFulfilledOrRejected` doesn't receive any arguments and its return value is ignored (unless it throws).